### ISSUE TITLE
Create export to Gridstack Voila (Dejavu?) HTML menu item?

### ISSUE THREAD

**Issue by bdklahn** [voila-dashboards/voila-gridstack#180]
Hello!
I love the gridstack template! I almost feel like it should be used as the default one for HTML export. It seems only natural to not want to limit display to only one column, for an html page.
Even old-school, primitive, web pages have sidebars, etc. :-)

But I understand the legacy of just wanting to clearly display the linear progression of a notebook exploration.

Anyway . . . I wonder how hard it might be to include a menu option to (apply the Voila/Dejavu settings and) export using the gridstack template. I mean . . . I guess one can download the displayed "breakout web page", but . . .

Here's how I now do it in a small "render_utils" library I created (which also handles papermill parameters cell injection).

```python
    c = Config()
    c.TemplateExporter.exclude_input = True
    c.TemplateExporter.exclude_output_prompt = True
    c.TemplateExporter.exclude_input_prompt = True
    c.ExecutePreprocessor.enabled=True
    c.HTMLExporter.template_name = 'gridstack'
```
(and I guess that might even be able to just inherit from the Voila template???)
Would it be too hard to add something like the Reveal.js one?

Thanks!

**Comment by hbcarlos** [comment]
Hi @bdklahn! Thank you for the kind words.
I believe the items on the export menu come from [nbconvet](https://github.com/jupyter/nbconvert), and Jupyterlab just uses nbconvert to export the notebook with the desired template.

We will need to move the template to the nbconvert project and then add the option in JupyterLab.

> PS. It occurs to me that maybe the Jupyter Lab repo is a more appropriate place to post this, but . . .

I think the place would be here and in nbconvert:
https://github.com/jupyter/nbconvert

**Comment by bdklahn** [comment]
Yes, I know that these templates are nbconvert-based, and (should) leverage that standard. But I didn't think that nbconvert maintained the graphical interfaces for manipulating and rendering these, nor was responsible for maintaining all the community-provided templates. I was/am thinking that, whoever maintains, say, the Jupyter Gridstack editor (which I assume must already utilize nbconvert rendering when it "pops out" a new browser window, with html), could fairly easily add another small graphical element in the form of an export menu entry. It seems like the entry would have to activate a very similar rendering config to that pop out, but maybe also offer the export location to be a server writable location (if/where available).
I thought maybe that was here.
But maybe it is not here where that editor is maintained . . . or maybe the menu items are considered more "core" additions to the Jupyter Lab UI.
Is it the case that nbconvert, in addition to offering rendering and the templating spec, is also responsible for what templates are "blessed" to have a place in that menu?
